### PR TITLE
Validation api

### PR DESCRIPTION
**Pull request by dczulada**
Pull requests into Cypress require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [ ] This pull request describes why these changes were made.
- [ ] Internal ticket for this PR:
- [ ] Internal ticket links to this PR
- [ ] Code diff has been done and been reviewed
- [ ] Tests are included and test edge cases
- [ ] Tests have been run locally and pass

**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: lib/validators/qrda_upload_validator.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    def qrda_1_validator(bundle_year, organization)
+      if bundle_year == 2020
+        organization == 'hl7' ? Cat1R52.instance : ::Validators::CMSQRDA1HQRSchematronValidator.new(bundle_year, false)
+      elsif bundle_year == 2021
+        organization == 'hl7' ? Cat1R52.instance : ::Validators::CMSQRDA1HQRSchematronValidator.new(bundle_year, false)
+      end
+    end
+
+    def qrda_3_validator(bundle_year, organization)
+      if bundle_year == 2020
```
Comment (lmd59) [inline lib/validators/qrda_upload_validator.rb:28]:
```suggestion
      if bundle_year == 2020 || bundle_year == 2021
```
remove elsif

File: app/controllers/qrda_uploads_controller.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  def index
+    respond_with(possible_qrda_uploaders)
+  end
+
+  def create
+    path = request.env['PATH_INFO'].split('.')[0]
+    uploader = possible_qrda_uploaders.select { |u| u[:path] == path }.first
+    if uploader
+      upload_task = QrdaUploadTask.new(year: params['year'], qrda_type: params['qrda_type'], organization: params['organization'])
```
[the remark this answers is not in the document]
Comment (lmd59) [reply app/controllers/qrda_uploads_controller.rb:15]: (i.e. giving/logging more specific errors?)

File: test/controllers/qrda_uploads_controller_test.rb [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+      assert_response 201, 'response should be Created on test_execution creation'
+      qrda_response = JSON.parse(response.body)
+      assert_not_nil qrda_response
+      assert qrda_response['execution_errors'][0]['message'].include? 'file count of 2 which exceeds upload limits of 20 MB and 1 files.'
+    end
+  end
+
+  # XML
+
+  test 'should get index with xml request' do
```
Comment (lmd59) [inline test/controllers/qrda_uploads_controller_test.rb:81]: Is this the right test name, since we're posting create instead of getting index?

**Review by lmd59 — APPROVED** [review]
